velox/connectors/hive/storage_adapters/hdfs/tests/HdfsFileSystemTest.cpp

```diff
@@ -350,11 +361,51 @@ TEST_F(HdfsFileSystemTest, multipleThreadsWithFileSystem) {
   }
 }
 
+TEST_F(HdfsFileSystemTest, write) {
+  std::string path = "/a.txt";
+  auto writeFile = openFileForWrite(path);
+  std::string data = "abcdefghijk";
+  writeFile->append(data);
+  writeFile->append(data);
+  writeFile->append(data);
+  writeFile->flush();
+  writeFile->close();
+  uint64_t file_size = writeFile->size();
+  ASSERT_EQ(file_size, data.size() * 3);
+}
+
+TEST_F(HdfsFileSystemTest, missingFileForWrite) {
+  const std::string filePath = "hdfs://localhost:7777/path/that/does/not/exist";
+  const std::string errorMsg =
+      "Failed to open hdfs file: hdfs://localhost:7777/path/that/does/not/exist";
+  VELOX_ASSERT_THROW(openFileForWrite(filePath), errorMsg);
+}
+
+TEST_F(HdfsFileSystemTest, writeDataFailures) {
+  const std::string errorMsg = "Write failure in HDFSWriteFile::append";
+  auto writeData = ([] {
+    auto writeFile = openFileForWrite("/a.txt");
+    writeFile->close();
+    writeFile->append("abcde");
+  });
+  VELOX_ASSERT_THROW(writeData(), errorMsg);
```

Comment:
I assume the exception is expected from append, but not from other code. If that's the case, write it like this:

```
   auto writeFile = openFileForWrite("/a.txt");
    writeFile->close();
    VELOX_ASSERT_THROW(writeFile->append("abcde"), errorMsg);
```